build.sbt

```diff
@@ -214,6 +215,33 @@ lazy val scip = project
   )
   .dependsOn(semanticdb, scipProto)
 
+lazy val mavenPlugin = project
+  .in(file("maven-plugin"))
+  .settings(
+    moduleName := "maven-plugin",
+    javaToolchainVersion := "8",
+    javaOnlySettings,
+    libraryDependencies ++=
+      Seq(
+        "org.apache.maven" % "maven-plugin-api" % "3.6.3",
+        "org.apache.maven.plugin-tools" % "maven-plugin-annotations" % "3.6.4" %
+          Provided,
+        "org.apache.maven" % "maven-project" % "2.2.1"
+      ),
+    Compile / resourceGenerators +=
```

Comment:
Maven has a special packaging mode for plugins, which seems to be nothing more than a post-processing step that introspects the plugin's pom.xml + all the Mojo files, then writes the special plugin.xml file.

https://maven.apache.org/guides/plugin/guide-java-plugin-development.html

Setting up a Maven build just for that seems to be counterproductive, so we just write the `plugin.xml` by hand using a template. It's obviously not great, but at this point I don't see a better solution, and we don't expect this plugin to change very frequently. 

`plugin.xml` will be force-kept in sync by tests, because Maven derives _a lot_ of information from it, so if it's broken - things won't work.

For local development this allows you to do `mavenPlugin/publishM2` and use that in maven builds.

Reply:
Makes sense. I love sbt